src/Symfony/Component/Dotenv/Command/DebugCommand.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Dotenv\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Dotenv\Dotenv;
+
+/**
+ * A console command to debug current dotenv files with variables and values.
+ *
+ * @author Christopher Hertel <mail@christopher-hertel.de>
+ */
+final class DebugCommand extends Command
+{
+    protected static $defaultName = 'debug:dotenv';
+    protected static $defaultDescription = 'Lists all dotenv files with variables and values';
+
+    private $kernelEnvironment;
+    private $projectDirectory;
+
+    public function __construct(string $kernelEnvironment, string $projectDirectory)
+    {
+        $this->kernelEnvironment = $kernelEnvironment;
+        $this->projectDirectory = $projectDirectory;
+
+        parent::__construct();
+    }
+
+    public function isEnabled(): bool
+    {
+        return \array_key_exists('SYMFONY_DOTENV_VARS', $_SERVER);
```

Comment:
Is that needed? Wouldn't it be surprising to have the command not work when .env files are empty?

Reply:
ref https://github.com/symfony/symfony/pull/42580#discussion_r690059976

the empty files ... i could not think of :clap:

Reply:
However, it does raise another question. Is the command surprising if Dotenv wasnt actually invoked?

Reply:
with empty files `SYMFONY_DOTENV_VARS=[]` (not undefined)? Then we're good IIUC.

Reply:
just checked, if .env files are empty there is still something like `APP_ENV` in `SYMFONY_DOTENV_VARS`. but if `Dotenv::load` was not called, the key doesn't exist. in that case im exiting with an error now